src/hyperion/experiment_plans/panda_flyscan_xray_centre_plan.py

```diff
@@ -22,6 +22,7 @@
 from hyperion.device_setup_plans.read_hardware_for_setup import (
     read_hardware_for_ispyb_during_collection,
     read_hardware_for_ispyb_pre_collection,
+    read_hardware_for_nexus_writer,
```

Comment:
These `read_hardware` methods are quite tightly coupled to the corresponding callbacks. Perhaps we should consider making it a bit more explicit, rather than dumping them all into one general-purpose module perhaps they should be in a separate sub-package in modules named after the corresponding callback, along with the necessary decorator/wrappers to register the CB, that way it's obvious to a consumer of the plans what is needed to use them.

Reply:
Thanks, I agree it's not clear. Made https://github.com/DiamondLightSource/hyperion/issues/1313 to discuss